### PR TITLE
Validation for creating/editing duplicate hospitals and ambulances

### PR DESCRIPTION
**Pull request by patrick-du**
[Notion ticket](https://www.notion.so/uwblueprintexecs/no-duplicate-resources-8713dd5852f04c0eb67558cf562da051)

Associated backend PR: https://github.com/uwblueprint/paramedics-web/pull/77

## Changes
- Added frontend validation to prevent the creation/editing of hospitals with the same `name` and ambulances with the same `vehicleNumber`
- Removed `GET_AMBULANCE/HOSPITAL_BY_ID` network call since we already cache the results of `GET_ALL_AMBULANCES/HOSPITALS` and a list of ambulances/hospitals is required in the mode `edit` to check if a duplicate exists

## Screenshots
Adding an existing hospital
image

Editing the hospital name to an existing hospital
image

Adding an existing ambulance
image

Editing the ambulance vehicleNumber to an existing ambulance
image

## Testing
- Create a hospital that already exists - result: modal should pop up
- Edit an existing hospital to another hospital that already exists - result: modal should pop up
- Create an ambulance that already exists - result: modal should pop up
- Edit an existing ambulance to another ambulance that already exists - result: modal should pop up
- Creating/editing hospitals and ambulances that do not already exist should work as expected

before opening PR
- [x] check notion ticket
- [x] run linter
- [x] go through file diff

filling out PR
- [x] descriptive title
- [x] update `notion ticket` link
- [ ] update `figma link`
- [x] fill out `Changes`
  - [ ] \(optional) add inline comments in diff
- [x] fill out `Testing`
- [x] \(optional) add `Screenshots`
- [x] assign yourself to the PR

after opening PR
- [x] link PR to notion ticket
- [x] move ticket to `In PR`
- [x] make sure build passes
- [x] ping `@ps` in `#paramedics-dev`

### CODE REVIEW

**Review by aaronabraham311 — DISMISSED** [review]
Nice work!

File: src/components/ResourceForm/AmbulanceFormPage.tsx [hunk 1]
```diff
@@ -73,14 +68,17 @@ const AmbulanceFormPage = ({
   });
 
   const [ambulanceNumber, setAmbulanceNumber] = useState<string>('');
+  const [openValidationError, setOpenValidationError] = useState<boolean>(
```
Comment (petorsong) [inline src/components/ResourceForm/AmbulanceFormPage.tsx]: nit: `isModalOpen` might be a little clearer and more concise in this context
```suggestion
  const [isModalOpen, setIsModalOpen] = useState<boolean>(
```

**Review by petorsong — DISMISSED** [review]
Ⓜ️

**Review by aaronabraham311 — APPROVED** [review]

**Review by petorsong — APPROVED** [review]
🦺

**Pull request merged**